lib/api/objectPutPart.js

```diff
@@ -87,80 +56,146 @@ export default function objectPutPart(authInfo, request, streamingV4Params,
+                    method: 'objectPutPart::metadata.getBucket',
+                });
+                return next(err);
+            }
+            return next(null, mpuBucket);
+        }),
+        // Check authorization of the MPU shadow bucket.
+        (mpuBucket, next) => {
+            // BACKWARD: Remove to remove the old splitter
+            if (mpuBucket.getMdBucketModelVersion() < 2) {
+                splitter = constants.oldSplitter;
+            }
+            const mpuOverviewKey = _getPrefixKey(splitter, objectKey, uploadId);
+            return metadata.getObjectMD(mpuBucketName, mpuOverviewKey, log,
+                (err, res) => {
+                    if (err) {
+                        log.error('error getting the object from mpu bucket', {
+                            error: err,
+                            method: 'objectPutPart::metadata.getObjectMD',
+                        });
+                        return next(err);
                     }
-                    // We pad the partNumbers so that the parts will be sorted
-                    // in numerical order
-                    const paddedPartNumber = `000000${partNumber}`.substr(-5);
-                    const serverSideEncryption =
-                              bucket.getServerSideEncryption();
-                    if (serverSideEncryption) {
-                        return kms.createCipherBundle(
-                            serverSideEncryption, log, (err, cipherBundle) => {
-                                services.metadataValidateMultipart(
-                                    metadataValMPUparams,
-                                    doStore(objectKeyContext, cipherBundle,
-                                            request, log, paddedPartNumber,
-                                            size, uploadId,
-                                            streamingV4Params, cb));
-                            });
+                    const initiatorID = res.initiator.ID;
+                    const requesterID = authInfo.isRequesterAnIAMUser() ?
+                        authInfo.getArn() : authInfo.getCanonicalID();
+                    if (initiatorID !== requesterID) {
+                        return next(errors.AccessDenied);
                     }
-                    return services.metadataValidateMultipart(
-                        metadataValMPUparams,
-                        doStore(objectKeyContext, null, request, log,
-                                paddedPartNumber, size, uploadId,
-                                streamingV4Params, cb));
+                    return next();
                 });
-        });
+        },
+        // Store in data backend.
+        next => {
+            const objectKeyContext = {
+                bucketName,
+                owner: canonicalID,
+                namespace: request.namespace,
+            };
+            return services.dataStore(null, objectKeyContext, cipherBundle,
```

Comment:
After `services.metadataStorePart` this would be the last services call left in this api. My recommendation is to move it to apiUtils/object. Then we could introduce a refactor for other apis to follow the suit.